components/Player/Player.jsx

```diff
@@ -1,10 +1,58 @@
 import React from 'react';
 import { PlayerMatchesTable } from '../Table';
+import PlayerHeader from './PlayerHeader';
 import Error from '../Error';
+import { getPlayer } from '../../actions';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import styles from './PlayerHeader.css';
 
-export default ({ params }) => (
-  <div>
-    {params && params.account_id && <PlayerMatchesTable playerId={params.account_id} numMatches={10} />}
-    {(!params || !params.account_id) && <Error />}
-  </div>
-);
+const Player = ({ params }) => {
+  if (!params || !params.account_id) {
+    return <Error />;
+  }
+
+  return (
+    <div>
+      <div className={styles.header}>
+        <PlayerHeader />
+      </div>
+      <PlayerMatchesTable playerId={params.account_id} numMatches={20} />
+    </div>
+  );
+};
+
+const mapStateToProps = (state, { params }) => ({ playerId: params.account_id });
+
+const mapDispatchToProps = (dispatch) => ({
+  getPlayer: (playerId) => dispatch(getPlayer(playerId)),
+});
+
+class RequestLayer extends React.Component {
+  componentDidMount() {
+    this.props.getPlayer(this.props.playerId);
+  }
+
+  componentWillUpdate(nextProps) {
+    if (this.props.playerId === nextProps.playerId) {
+      this.props.getPlayer(nextProps.playerId);
+    }
+  }
+
+  render() {
+    return <Player {...this.props} />;
+  }
+}
+
+export default withRouter(connect(mapStateToProps, mapDispatchToProps)(RequestLayer));
```

Comment:
withRouter just passes any router props down to it, it's just a different way of doing it. Not sure which is better but want to try to avoid passing props down a lot of components if we can avoid it (just learned about this method)